src/generic_ui/scripts/ui.ts

```diff
@@ -190,11 +191,27 @@ module UI {
      * Upon construction, the UI installs update handlers on core.
      */
     constructor(
-        public core   :uproxy_core_api.CoreApi,
+        public core   :CoreConnector,
         public browserApi :BrowserAPI) {
       // TODO: Determine the best way to describe view transitions.
       this.view = ui_constants.View.SPLASH;  // Begin at the splash intro.
 
+      core.on('connect', () => {
```

Comment:
I think that this is a completely correct naming, they represent effectively when the core connector is connected and disconnected.  I would personally call them more than chrome specific: they express when we are actually connected to the core environment.  At the moment, with having Chrome and Firefox, we end up with the only use being in Chrome (in Firefox, connecting is essentially a noop), it seems like a good thing to have in general though.